### PR TITLE
Add lint-action and action-eslint

### PR DESCRIPTION
**Pull request by Charl1996**
This adds the new linter(s) to the repo, lint-action and action-eslint.

See [this PR](https://github.com/dimagi/commcare-hq/pull/33280) as example usage. 

In addition to the linked PR I also verified the following scenarios:

1. No .py or .js file changes should not run the linters (the jobs does run until it determines there's not relevant files to lint over).
2. If a linter is not run it should green check that particular linting job.
3. If only .js files changed only eslint should run and flake8 should not be invoked (green check it by default).
4. If only .py files changed only flake8 should run and eslint should not be invoked (green check it by default).
5. If a linter fails the PR can still be merged.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        uses: tj-actions/changed-files@v24
+        with:
+          files: |
+            **/*.py
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        with:
+          python-version: 3.8
```
Comment (millerdev) [inline .github/workflows/lint.yml:34]: Why 3.8? HQ currently requires 3.9.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Install Python dependencies
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: pip install flake8
+
+      - name: Run flake8 linter
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        uses: wearerequired/lint-action@v2
+        with:
```
Comment (esoergel) [inline .github/workflows/lint.yml:43]: I assume this auto-discovers the `.flake8` config file (and likewise for javascript)?

**Review by esoergel — APPROVED** [review]
Thanks!

File: .github/workflows/lint.yml (hunk 2 above)
Comment (Charl1996) [reply .github/workflows/lint.yml:43]: Correct.

**Review by millerdev — APPROVED** [review]

**Pull request merged**